packages/app-builder-lib/src/util/appFileCopier.ts

```diff
@@ -191,25 +208,38 @@ function validateFileSet(fileSet: ResolvedFileSet): ResolvedFileSet {
 
 /** @internal */
 export async function computeNodeModuleFileSets(platformPackager: PlatformPackager<any>, mainMatcher: FileMatcher): Promise<Array<ResolvedFileSet>> {
-  const deps = await platformPackager.info.getNodeDependencyInfo(platformPackager.platform).value
+  const deps = (await platformPackager.info.getNodeDependencyInfo(platformPackager.platform).value) as Array<NodeModuleInfo>
+
   const nodeModuleExcludedExts = getNodeModuleExcludedExts(platformPackager)
   // serial execution because copyNodeModules is concurrent and so, no need to increase queue/pressure
   const result = new Array<ResolvedFileSet>()
   let index = 0
   for (const info of deps) {
-    const source = info.dir
+    const source = platformPackager.info.appDir + path.sep + "node_modules" + path.sep + info.name
+    // const source = platformPackager.info.appDir
     const destination = getDestinationPath(source, { src: mainMatcher.from, destination: mainMatcher.to, files: [], metadata: null as any })
 
     // use main matcher patterns, so, user can exclude some files in such hoisted node modules
     // source here includes node_modules, but pattern base should be without because users expect that pattern "!node_modules/loot-core/src{,/**/*}" will work
-    const matcher = new FileMatcher(path.dirname(source), destination, mainMatcher.macroExpander, mainMatcher.patterns)
+    const matcher = new FileMatcher(path.dirname(platformPackager.info.appDir + path.sep + "node_modules"), destination, mainMatcher.macroExpander, mainMatcher.patterns)
     const copier = new NodeModuleCopyHelper(matcher, platformPackager.info)
-    const files = await copier.collectNodeModules(
-      source,
-      info.deps.map(it => it.name),
-      nodeModuleExcludedExts
-    )
+    const files = await copier.collectNodeModules(info, nodeModuleExcludedExts)
     result[index++] = validateFileSet({ src: source, destination, files, metadata: copier.metadata })
+
+    if (info.conflictDependency) {
+      for (const dep of info.conflictDependency) {
+        const source = platformPackager.info.appDir + path.sep + "node_modules" + path.sep + info.name + path.sep + "node_modules" + path.sep + dep.name
```

Comment:
Curious, how do we know that it's a double-nested `node_modules`? I'm worried about the hard-coded approach here, but wanted to understand the context here.

Reply:
All dependencies are placed directly into a map. If it's found that there's already a dependency with the same name but a different version in the map, it indicates a conflict. Currently, I'm placing the conflicting version directly into the node_modules of the current package.
https://github.com/beyondkmp/app-builder/blob/c85c25c911b6f5fc508afbbbd705b37220571b12/pkg/node-modules/nodeModuleCollector.go#L91-L107

The name conflictDependency might be somewhat misleading; we could consider renaming it.